src/diagnostics.rs

```diff
@@ -279,6 +287,12 @@ fn report_msg<'tcx>(
         err.note(&span_msg);
         err.note("(no span available)");
     }
+
+    // Add extra labels to the primary span.
+    for label in labels {
+        err.span_label(span, label);
+    }
```

Comment:
Further up we have some extra logic to make sure the `span_msg` is visible even on a `DUMMY_SP`... that should probably also be taken into account for the extra labels?

Reply:
Yes, it probably should. Do you have any examples of cases where we can't get a span for the primary error? Things will probably look pretty chaotic anyway, but it would be nice to see.

Reply:
I'm afraid I don't know any examples. :/

Reply:
Well, I haven't run into any so I'm going to declare this rare enough to not worry too much about for the moment.

Reply:
Yeah, as long as the existing logic is roughly preserved I think this is fine.